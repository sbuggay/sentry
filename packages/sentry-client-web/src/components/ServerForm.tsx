import * as React from "react";

interface IServerFormProps {
    handleSubmit?: (state: IServerFormState) => any;
    onSubmit?: () => any;
}

interface IServerFormState {
    nameValue?: string;
    hostValue?: string;
}

export default class ServerForm extends React.Component<IServerFormProps & IServerFormState, any> {
    constructor(props: any) {
        super(props);

        this.state = {
            nameValue: "",
            hostValue: ""
        };
    }

    public handleChange(event: any) {
        const target = event.target;
        const value = target.type === "checkbox" ? target.checked : target.value;
        const name = target.name;
        this.setState({ [name]: value });
    }

    public handleSubmit(event: any) {
        if (this.props.handleSubmit) {
            this.props.handleSubmit(this.state);
        }
        event.preventDefault();
    }

    public render() {
        return (
            <form onSubmit={this.handleSubmit.bind(this)}>
                <input
                    placeholder="name"
                    type="text"
                    name="nameValue"
                    value={this.state.nameValue}
                    onChange={this.handleChange.bind(this)}>
                </input>
                <input
                    placeholder="host"
                    type="text"
                    name="hostValue"
                    value={this.state.hostValue}
                    onChange={this.handleChange.bind(this)}>
                </input>
                <button type="submit">submit</button>
            </form>
        );
    }
}
