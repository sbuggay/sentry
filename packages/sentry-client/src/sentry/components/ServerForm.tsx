import * as React from "react";

interface IProps {
	handleSubmit?: Function;
	onSubmit?: Function;
};

export default class ServerForm extends React.Component<IProps, any> {
	constructor(props: any) {
		super(props);

		this.state = {
			nameValue: "",
			hostValue: ""
		}
	}

	handleChange(event: any) {
		const target = event.target;
		const value = target.type === 'checkbox' ? target.checked : target.value;
		const name = target.name;
		this.setState({ [name]: target.value });
	}

	handleSubmit(event: any) {
		this.props.handleSubmit(this.state);
		event.preventDefault();
	}

	render() {
		return (
			<form onSubmit={this.handleSubmit.bind(this)}>
				<input placeholder="name" type="text" name="nameValue" value={this.state.nameValue} onChange={this.handleChange.bind(this)}></input>
				<input placeholder="host" type="text" name="hostValue" value={this.state.hostValue} onChange={this.handleChange.bind(this)}></input>
				<button type="submit">submit</button>
			</form>
		);
	}
};
