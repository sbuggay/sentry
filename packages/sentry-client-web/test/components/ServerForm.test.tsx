import * as React from "react";
import * as renderer from "react-test-renderer";
import ServerForm from "../../src/components/ServerForm";

describe("ServerForm", () => {
	it("renders a ServerForm", () => {
		const component = renderer.create(
			<ServerForm />
		);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
