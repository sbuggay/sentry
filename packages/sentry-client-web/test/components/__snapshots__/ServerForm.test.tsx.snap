// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerForm renders a ServerForm 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="nameValue"
    onChange={[Function]}
    placeholder="name"
    type="text"
    value=""
  />
  <input
    name="hostValue"
    onChange={[Function]}
    placeholder="host"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    submit
  </button>
</form>
`;
